specs/ExecuteScriptWithResult.md

```diff
@@ -0,0 +1,315 @@
+                        // such as `throw 1`.
+                        wil::unique_cotaskmem_string exceptionName;
+                        exception->get_Name(&exceptionName);
+
+                        // The ExceptionMessage property could be the empty string if script throws a non-Error object, 
+                        // such as `throw 1`.
+                        wil::unique_cotaskmem_string exceptionMessage;
+                        exception->get_Message(&exceptionMessage)
+
+                        // Get the location of the exception, note that the coordinates
+                        // here are 0 as the starting position.
+                        uint32_t lineNumber = 0;
+                        uint32_t columnNumber = 0;
+                        exception->get_LineNumber(&lineNumber);
+                        exception->get_ColumnNumber(&columnNumber);
+
+                        auto exceptionInfo = 
+                                L"The script execution failed." + 
+                                L"\nName: " + exceptionName.get() +
+                                L"\nMessage: " + exceptionMessage.get() +
+                                L"\nLineNumber: " + std::to_wstring(lineNumber) +
+                                L", ColumnNumber:" + std::to_wstring(columnNumber);
+                        MessageBox(
+                            nullptr, exceptionInfo.c_str(),
+                            L"ExecuteScript Result", MB_OK);
+                    }
+                }
+                return S_OK;
+            })
+            .Get());
+}
+
+```
+## .NET and WinRT
+```c#
+class ExecuteScriptWithResultDemo {
+    int idx = 0;
+
+    private String GenerateScriptCode(String str, String reg, String item) {
+        String ret = "let str_" + idx + " = \"" + str + "\"; let n_" + idx + "= str_" + idx 
+            + ".replace(" + reg + ", \"" + item + "\"); n_" + idx + ";";
+        ++idx;
+        return ret;
+    }
+
+
+    // This is a demo that uses regular expressions in 
+    // javascript to complete string replacement, it will handle
+    // the case of successful execution and execution exception
+    public void MatchRegWithScript(String str, String reg, String item) {
+        var environment = webView2Control.CoreWebView2.Environment;
+        String script = GenerateScriptCode(str, reg, item);
+        CoreWebView2ExecuteScriptResult result = await ExecuteScriptWithResultAsync(script);
+        
+        bool isSuccess = result.IsSuccess;
+        // Here is the successful execution.
+        if (isSuccess) {            
+            // Try to get the string result, it will throw an exception
+            // if the result type isn't string type.
+            try {
+                String stringResult = result.TryGetResultAsString();
+                Debug.WriteLine($"replaced string: {result.stringResult}");
+            }
+            catch (ArgumentException) {
```

Comment:
It is doc'd in the member descriptions, but it shouldn't be throwing per other comment discussions.

Reply:
Since the design of the `TryGetResultAsString` above, does it need to be modified here?

btw the interface design here refers to [TryGetWebMessageAsString](https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2webmessagereceivedeventargs?view=webview2-1.0.1150.38#trygetwebmessageasstring)